src/features/shared-link-settings-modal/SharedLinkSettingsModal.js

```diff
@@ -15,8 +15,31 @@ import ExpirationSection from './ExpirationSection';
 import AllowDownloadSection from './AllowDownloadSection';
 import messages from './messages';
 
+import { PEOPLE_WITH_LINK, PEOPLE_IN_COMPANY, PEOPLE_IN_ITEM } from '../shared-link-modal/constants';
+
 import './SharedLinkSettingsModal.scss';
 
+/**
+ * Return the translation key based on the access level and whether the user can download
+ * @param {string} accessLevel one of 'peopleWithTheLink', 'peopleInYourCompany', or 'peopleInThisItem'
+ * @param {boolean} canDownload prop value for whether the user can currently download
+ *
+ * @return {string} message ID for the proper transloation (or empty string when nothing matches)
+ */
+function getAccessNoticeMessageId(accessLevel, canDownload) {
+    switch (accessLevel) {
+        case PEOPLE_WITH_LINK:
+            return canDownload ? 'withLinkViewDownload' : 'withLinkView';
+        case PEOPLE_IN_COMPANY:
+            return canDownload ? 'inCompanyViewDownload' : 'inCompanyView';
+        case PEOPLE_IN_ITEM:
+            return 'inItem';
+        default:
+            // return falsy meaning 'there is no key to return'
+            return '';
```

Comment:
Should it ever be here?

Reply:
it shouldn't, but we have some lint rule that requires adding `default`